yellowbrick/features/importances.py

```diff
@@ -0,0 +1,250 @@
+# yellowbrick.features.importances
+# Feature importance visualizer
+#
+# Author:  Benjamin Bengfort <benjamin@bengfort.com>
+# Created: Fri Mar 02 15:21:36 2018 -0500
+#
+# Copyright (C) 2018 District Data Labs
+# For license information, see LICENSE.txt
+#
+# ID: importances.py [] benjamin@bengfort.com $
+
+"""
+Implementation of a feature importances visualizer. This visualizer sits in
+kind of a weird place since it is technically a model scoring visualizer, but
+is generally used for feature engineering.
+"""
+
+##########################################################################
+## Imports
+##########################################################################
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+from yellowbrick.utils import is_dataframe
+from yellowbrick.base import ModelVisualizer
+from yellowbrick.exceptions import YellowbrickTypeError, NotFitted
+
+
+##########################################################################
+## Feature Visualizer
+##########################################################################
+
+class FeatureImportances(ModelVisualizer):
+    """
+    Displays the most informative features in a model by showing a bar chart
+    of features ranked by their importances. Although primarily a feature
+    engineering mechanism, this visualizer requires a model that has either a
+    ``coef_`` or ``feature_importances_`` parameter after fit.
+
+    Parameters
+    ----------
+    model : Estimator
+        A Scikit-Learn estimator that learns feature importances. Must support
+        either ``coef_`` or ``feature_importances_`` parameters.
+
+    ax : matplotlib Axes, default: None
+        The axis to plot the figure on. If None is passed in the current axes
+        will be used (or generated if required).
+
+    labels : list, default: None
+        A list of feature names to use. If a DataFrame is passed to fit and
+        features is None, feature names are selected as the column names.
+
+    relative : bool, default: True
+        If true, the features are described by their relative importance as a
+        percentage of the strongest feature component; otherwise the raw
+        numeric description of the feature importance is shown.
+
+    kwargs : dict
+        Keyword arguments that are passed to the base class and may influence
+        the visualization as defined in other Visualizers.
+
+    Examples
+    --------
+
+    >>> from sklearn.ensemble import GradientBoostingClassifier
+    >>> visualizer = FeatureImportances(GradientBoostingClassifier())
+    >>> visualizer.fit(X, y)
+    >>> visualizer.poof()
```

Comment:
a thought, do we want the fit method to call poof()? it would be one less step to produce the visualization

Reply:
That's a good thought, the problem is that poof calls `plt.show()` and we need it to not do that for image comparison and other reasons. However, poof also calls `finalize` and I'm thinking that should be part of `draw()` not part of poof. I'll think about how to do this.

Reply:
Maybe we want to add a `fit_poof` method similar to sklearn's `fit_transform`?

Reply:
That's a good idea, I thought we had that but they don't seem to be there anymore.